ding_periodical.module

```diff
@@ -135,12 +136,17 @@ function theme_ding_periodical_issues($variables) {
 
       // Check if the volume is reservable and add reservation button.
       if (!empty($issue_id) && ding_periodical_is_reservable($issues[$key])) {
-        $path = array($issue_id, $volume, $key);
+        $item_id = array($issue_id, $volume, $key);
+
+        foreach ($item_id as &$v) {
+          $v = rawurlencode($v);
```

Comment:
Thank you for elaborating.

If `rawurlencode()` is only required for communication with the LMS and not related to the `hook_menu()`  then the encoding should be pushed down the call stack as mentioned in the first comment.

Since this problem has only been identified for DDELibra/Alma then placing the fix in the Alma specific code seems like the right approach to me. This could either be [in the provider](https://github.com/ding2/alma/blob/master/includes/alma.reservation.inc#L269) or [the client class](https://github.com/ding2/alma/blob/master/lib/AlmaClient/AlmaClient.class.php#L441). If we keep the encoding at this level the change will affect all providers even though they do not suffer from the same problem.